quarkus/service/src/main/java/org/apache/polaris/service/quarkus/config/ProductionReadinessChecks.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.polaris.service.context.RealmContextResolver;
+import org.apache.polaris.service.context.TestRealmContextResolver;
+import org.apache.polaris.service.persistence.InMemoryPolarisMetaStoreManagerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class ProductionReadinessChecks {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProductionReadinessChecks.class);
+
+  public void checkAuthenticator(
+      @Observes StartupEvent event,
+      Authenticator<String, AuthenticatedPolarisPrincipal> authenticator) {
+    if (authenticator instanceof TestInlineBearerTokenPolarisAuthenticator) {
+      warnOnTestOnly("authenticator", "polaris.authentication.authenticator.type");
+    }
+  }
+
+  public void checkTokenService(@Observes StartupEvent event, IcebergRestOAuth2ApiService service) {
+    if (service instanceof TestOAuth2ApiService) {
+      warnOnTestOnly("token service", "polaris.authentication.token-service.type");
+    }
+  }
+
+  public void checkTokenBroker(
+      @Observes StartupEvent event,
+      AuthenticationConfiguration configuration,
+      TokenBrokerFactory factory) {
+    if (factory instanceof JWTRSAKeyPairFactory) {
+      LOGGER.warn(
+          "No public and private key files were provided; these will be generated. "
+              + "Do not do this in production (offending configuration options: "
+              + "'polaris.authentication.token-broker.rsa-key-pair.public-key-file' and "
+              + "'polaris.authentication.token-broker.rsa-key-pair.private-key-file').");
+    }
+    if (factory instanceof JWTSymmetricKeyFactory) {
+      if (configuration
+          .tokenBroker()
+          .symmetricKey()
+          .map(SymmetricKeyConfiguration::secret)
+          .isPresent()) {
+        LOGGER.warn(
+            "A symmetric key secret was provided through configuration rather than through a secret file. "
+                + "Do not do this in production (offending configuration option: "
+                + "'polaris.authentication.token-broker.symmetric-key.secret').");
+      }
+    }
+  }
+
+  public void checkMetastore(@Observes StartupEvent event, MetaStoreManagerFactory factory) {
+    if (factory instanceof InMemoryPolarisMetaStoreManagerFactory) {
+      warnOnTestOnly("metastore", "polaris.persistence.type");
+    }
+  }
+
+  public void checkRealmResolver(
+      @Observes StartupEvent event,
+      RealmContextConfiguration configuration,
+      RealmContextResolver resolver) {
+    if (resolver instanceof TestRealmContextResolver) {
+      warnOnTestOnly("realm context resolver", "polaris.realm-context.type");
+    }
+    if (resolver instanceof DefaultRealmContextResolver) {
+      if (!configuration.requireHeader()) {
+        LOGGER.warn(
+            "The realm context resolver is configured to map requests without a realm header to the default realm. "
```

Comment:
I think this warning is justified only when using configuration defaults. If a user explicitly allows requests without a realm header, it should be fine... I'm not sure we can detect this, though :thinking:

Reply:
I did as you suggested, although I think we could argue that it is still a security risk, regardless of whether the user is doing it consciously or not.

Reply:
I completely agree that it's risk, but I think it would be unfair to keep WARNing on something that the user consciously configured. I assume, the act of setting that configuration property indicated taking responsibility :)